tool/python/singa/model.py

```diff
@@ -0,0 +1,362 @@
+
+    # deal with label layer (depreciated)
+    if self.label == True:
+      label_layer = Layer(name='label', type=kLabel)      
+      ly = net.layer.add()
+      ly.CopyFrom(label_layer.layer)
+      getattr(ly, 'srclayers').append(self.layers[0].layer.name)
+      getattr(lastly, 'srclayers').append(label_layer.layer.name)
+    else:
+      if lastly.name == 'RBMVis'+str(len(self.layers)-2):
+        getattr(lastly, 'srclayers').append(bly.name)
+      else:
+        getattr(lastly, 'srclayers').append(self.layers[0].layer.name)
+
+    setval(self.jobconf, neuralnet=net)
+
+  def fit(self, **kwargs):
+    pass
+
+  def evaluate(self, **kwargs):
+    pass
+    
+  def display(self):
+    print text_format.MessageToString(self.jobconf)
+
+
+class Sequential(Model):
+  def __init__(self, name='my model', argv=[], label=False):
+    super(Sequential, self).__init__(name=name, argv=argv, label=label)
+
+  def exist_datalayer(self, phase):
+    for ly in self.layers:
+      if ly.layer.include == enumPhase(phase):
+        return True
+    return False
+    
+  def fit(self, data=None, alg='bp', train_steps=0, **fields):
+    '''
+    required
+      data        = (Data)   // Data class object for training data
+      alg         = (string) // algorithm, e.g., 'bp', 'cd'
+      train_steps = (int)    // the number of training, i.e., epoch
+    optional
+      **fields (KEY=VALUE)
+        batch_size       = (int)    // batch size for training data
+        disp_freq        = (int)    // frequency to display training info
+        disp_after       = (int)    // display after this number 
+        validate_data    = (Data)   // validation data, specified in load_data()
+        validate_freq    = (int)    // frequency of validation
+        validate_steps   = (int)    // total number of steps for validation
+        validate_after   = (int)    // start validation after this number
+        checkpoint_path  = (string) // path to checkpoint file
+        checkpoint_freq  = (int)    // frequency for checkpoint
+        checkpoint_after = (int)    // start checkpointing after this number
+    '''
+    assert data != None, 'Training data shold be set'
+    assert train_steps != 0, 'Training steps shold be set'
+
+    if 'batch_size' in fields:  # if new value is set, replace the batch_size
+      setval(data.layer.store_conf, batchsize=fields['batch_size'])
+
+    if self.exist_datalayer('train') == False: 
+      self.layers.insert(0, data)
+
+    if 'validate_data' in fields:
+      self.layers.insert(1, fields['validate_data'])
+
+    setval(self.jobconf, train_steps=train_steps)
+    setval(self.jobconf, **fields)
+
+    # loading checkpoint if it is set
+    if data.checkpoint != None:
+      setval(self.jobconf, checkpoint_path=data.checkpoint)
+    
+    # set Train_one_batch component, using backprogapation at default
+    setval(self.jobconf, train_one_batch=Algorithm(type=enumAlgType(alg)).proto)
```

Comment:
fit() does not call Driver::Train()?
only evaluate calls Driver::Train().
If user_model.py does not have evaluate(), then the training would not be started.

there are three datasets involved in DL training, training dataset, validation dataset and test dataset.
fit() runs against training (and validation) dataset, whereas evaluate runs against test dataset.
fit() should be independently to evaluate().

Here, we may not be able to follow keras API exactly.